src/Components/Components/src/Routing/LegacyRouteMatching/LegacyOptionalTypeRouteConstraint.cs

```diff
@@ -1,20 +1,20 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-namespace Microsoft.AspNetCore.Components.Routing
+namespace Microsoft.AspNetCore.Components.LegacyRouteMatching
```

Comment:
To minimize any risk that the two different route matching implementations interfere with one another, I've put them in separate peer-level namespaces, neither of which references the other.

I've also renamed all the legacy ones to `Legacy***` to avoid any risk that, in the future, somebody references one from the legacy namespace by mistake via the "Add using statement" codefix.

Reply:
Also, I know it's ugly having all this extra code. But as soon as this patch is merged to `master`, we can also do a follow-up change in `master` that removes everything from the `LegacyRouteMatching` directory and its corresponding tests, so we pretty much won't see this in all future work.

Reply:
Sounds good to me, I would have done the same thing